stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,13 @@
+from typing import Any
+
+class Pattern:
+    def __getattr__(self, name: str) -> Any: ...  # incomplete
+
+class Match:
+    def __getattr__(self, name: str) -> Any: ...  # incomplete
+
+class Scanner:
+    def __getattr__(self, name: str) -> Any: ...  # incomplete
+
+class Splitter:
+    def __getattr__(self, name: str) -> Any: ...  # incomplete
```

Comment:
These objects are created in [_regex.c](https://github.com/mrabarnett/mrab-regex/blob/79d633b2dc9899e0f0cbb9ffdb81c14a4f7a86f2/regex_3/_regex.c#L533).

Is `_regex.pyi` the right place to try and annotate these classes?
They are returned by some of the public functions in `regex.py` so makes sense to have them typed.
(N.B. haven't annotated any of the methods yet as want to confirm this is the right approach first)

One issue I get with this is the stubtest warns that these are not available at runtime:
```
error: regex._regex.Pattern is not present at runtime
Stub: at line 3
<TypeInfo regex._regex.Pattern>
Runtime:
MISSING
```
Which is true if you try and import them:
![image](https://user-images.githubusercontent.com/80432516/147499094-50d88aab-8559-452f-8f2e-a98376310486.png)
but if you check the type of the returned Pattern object that's where it seems to imply that it's from:
![image](https://user-images.githubusercontent.com/80432516/147499170-2227b12b-9df3-42a5-9dc1-96cb27edea6f.png)

Reply:
> These objects are created in [_regex.c](https://github.com/mrabarnett/mrab-regex/blob/79d633b2dc9899e0f0cbb9ffdb81c14a4f7a86f2/regex_3/_regex.c#L533).
> 
> Is `_regex.pyi` the right place to try and annotate these classes?

While in the past we sometimes didn't bother in typeshed, doing so is the right thing and nowadays we try to follow the implementation structure if possible.

Reply:
The classes don't say `regex._regex.Pattern` though, and there is no module `_regex`. I would probably just put these in `regex/__init__.pyi`, but `regex/_regex.pyi` doesn't seem bad either.

Reply:
So rather confusingly there are two classes called `Scanner`. One is in `regex/_regex.c` and isn't exported in `__all__` in `__init__.py`. The other is defined `regex/_regex_core.py` and is exported in `__all__` in `__init__.py`.

To avoid confusion my preference would be to keep all the C classes together in `regex/_regex.pyi` and save `__init__.py` for the types exported by the package.